File: spring-server/src/main/java/me/pjq/model/SensorStatus.java
package me.pjq.model;

/**
 * Created by i329817(Jianqing.Peng@sap.com) on 16/10/2017.
 */
public class SensorStatus {
    public float distance;

    public boolean relay_on;
    /**
     * obstacles : {"obstacle1":false,"obstacle2":false,"obstacle3":true,"obstacle4":false}
     */

    public Obstacles obstacles;

    public static class Obstacles {
        /**
         * obstacle1 : false
         * obstacle2 : false
         * obstacle3 : true
         * obstacle4 : false
         */

        public boolean obstacle1;
        public boolean obstacle2;
        public boolean obstacle3;
        public boolean obstacle4;

        @Override
        public String toString() {
            return "Obstacles{" +
                    "obstacle1=" + obstacle1 +
                    ", obstacle2=" + obstacle2 +
                    ", obstacle3=" + obstacle3 +
                    ", obstacle4=" + obstacle4 +
                    '}';
        }
    }

    @Override
    public String toString() {
        return "SensorStatus{" +
                "distance=" + distance +
                ", obstacles=" + obstacles +
                '}';
    }
}

File: android/app/src/main/java/me/pjq/rpicar/models/SensorStatus.kt
package me.pjq.rpicar.models

/**
 * Created by i329817(Jianqing.Peng@sap.com) on 16/10/2017.
 */

class SensorStatus {
    var distance: Float = 0.toFloat()
    /**
     * obstacles : {"obstacle1":false,"obstacle2":false,"obstacle3":true,"obstacle4":false}
     */

    var relay_on: Boolean = false

    var obstacles: Obstacles? = null

    class Obstacles {
        /**
         * obstacle1 : false
         * obstacle2 : false
         * obstacle3 : true
         * obstacle4 : false
         */

        var obstacle1: Boolean = false
        var obstacle2: Boolean = false
        var obstacle3: Boolean = false
        var obstacle4: Boolean = false

        override fun toString(): String {
            return "obstacle1: " + obstacle1 +
                    " obstacle2: " + obstacle2 +
                    " obstacle3: " + obstacle3 +
                    " obstacle4: " + obstacle4
        }
    }

    override fun toString(): String {
        return "SensorStatus{" +
                "distance=" + distance +
                ", obstacles=" + obstacles +
                '}'
    }
}
